### PR TITLE
Fix doxygen warnings

### LINKED ISSUE #6583: Orth_tree: Incorrect documentation
Doing some extra tests on the CGAL documentation (setting `GENERATE_XML=YES`), the following warning appeared:
```
.../Orthtree/include/CGAL/Orthtree.h:860: warning: argument 'search_point' of command @param is not found in the argument list of CGAL::Orthtree< Traits_, PointRange_, PointMap_ >::nearest_k_neighbors_in_radius(Sphere &query_sphere, std::size_t k, OutputIterator output) const
.../Orthtree/include/CGAL/Orthtree.h:860: warning: argument 'search_radius_squared' of command @param is not found in the argument list of CGAL::Orthtree< Traits_, PointRange_, PointMap_ >::nearest_k_neighbors_in_radius(Sphere &query_sphere, std::size_t k, OutputIterator output) const
.../Orthtree/include/CGAL/Orthtree.h:860: warning: The following parameter of CGAL::Orthtree::nearest_k_neighbors_in_radius(Sphere &query_sphere, std::size_t k, OutputIterator output) const is not documented:
  parameter 'query_sphere'
```
Even though the function is not part of the official documentation the documentation of the function should be correct.

### LINKED ISSUE #6584: Poisson_surface_reconstruction_3: Incorrect documentation
Doing some extra tests on the CGAL documentation (setting `GENERATE_XML=YES`), the following warning appeared:
```
.../Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h:766: warning: Found unknown command '@commentheading'
.../Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h:764: warning: argument 'SparseLinearAlgebraTraits_d' of command @param is not found in the argument list of CGAL::Poisson_reconstruction_function< Gt >::solve_poisson(SparseLinearAlgebraTraits_d solver, double lambda)
.../Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h:764: warning: The following parameter of CGAL::Poisson_reconstruction_function::solve_poisson(SparseLinearAlgebraTraits_d solver, double lambda) is not documented:
  parameter 'lambda'
.../Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h:1206: warning: Found unknown command '@commentheading'
.../Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h:1206: warning: argument 'SparseLinearAlgebraTraits_d' of command @param is not found in the argument list of CGAL::Poisson_reconstruction_function< Gt >::assemble_poisson_row(typename SparseLinearAlgebraTraits_d::Matrix &A, Vertex_handle vi, typename SparseLinearAlgebraTraits_d::Vector &B, double lambda)
.../Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h:1206: warning: The following parameters of CGAL::Poisson_reconstruction_function::assemble_poisson_row(typename SparseLinearAlgebraTraits_d::Matrix &A, Vertex_handle vi, typename SparseLinearAlgebraTraits_d::Vector &B, double lambda) are not documented:
  parameter 'A'
  parameter 'vi'
  parameter 'B'
  parameter 'lambda'
.../Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h:351: warning: Found unknown command '@commentheading'
.../Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h:349: warning: argument 'InputIterator' of command @param is not found in the argument list of CGAL::Reconstruction_triangulation_3< BaseGt, Gt, Tds_ >::insert(InputIterator first, InputIterator beyond, PointPMap point_pmap, NormalPMap normal_pmap, Visitor visitor)
.../Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h:349: warning: argument 'PointPMap' of command @param is not found in the argument list of CGAL::Reconstruction_triangulation_3< BaseGt, Gt, Tds_ >::insert(InputIterator first, InputIterator beyond, PointPMap point_pmap, NormalPMap normal_pmap, Visitor visitor)
.../Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h:349: warning: argument 'NormalPMap' of command @param is not found in the argument list of CGAL::Reconstruction_triangulation_3< BaseGt, Gt, Tds_ >::insert(InputIterator first, InputIterator beyond, PointPMap point_pmap, NormalPMap normal_pmap, Visitor visitor)
.../Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h:349: warning: The following parameter of CGAL::Reconstruction_triangulation_3::insert(InputIterator first, InputIterator beyond, PointPMap point_pmap, NormalPMap normal_pmap, Visitor visitor) is not documented:
  parameter 'visitor'
.../Poisson_surface_reconstruction_3/include/CGAL/IO/output_surface_facets_to_triangle_soup.h:29: warning: Found unknown command '@commentheading'
```
Even though the function is not part of the official documentation the documentation of the function should be correct.

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Address Issue #6583, #6584,  and #6585 

## Release Management

* Affected package(s): Poisson Surface Reconstruction,  Polyhedron_3
* Issue(s) solved (if any): fix #6583, #6584, #6585
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

File: Polyhedron/include/CGAL/Polyhedron_3.h [hunk 1]
```diff
@@ -1448,10 +1448,10 @@ class Polyhedron_3 {
 
     /// Erases the small connected components and the isolated vertices.
     ///
-    /// @commentheading Preconditions:
+    /// *Preconditions:*
```
Comment (albert-github) [inline Polyhedron/include/CGAL/Polyhedron_3.h:1451]: Why not use the doxygen `\pre` command?

File: Surface_mesher/include/CGAL/IO/output_surface_facets_to_polyhedron.h [hunk 2]
```diff
@@ -23,17 +23,17 @@
...
 ///
 /// This variant exports the surface as a polyhedron.
 /// It requires the surface to be manifold. For this purpose,
-/// you may call make_surface_mesh() with Manifold_tag or Manifold_with_boundary_tag parameter.
+/// you may call `make_surface_mesh()` with `Manifold_tag` or `Manifold_with_boundary_tag` parameter.
 ///
-/// @commentheading Template Parameters:
-/// @param SurfaceMeshComplex_2InTriangulation_3 model of the SurfaceMeshComplex_2InTriangulation_3 concept.
-/// @param Polyhedron an instance of CGAL::Polyhedron_3<Traits>.
+/// *Template Parameters:*
```
Comment (albert-github) [inline Surface_mesher/include/CGAL/IO/output_surface_facets_to_polyhedron.h:32]: Is this line not a bit overdone as doxygen will add a header **Template Parameters**

File: Poisson_surface_reconstruction_3/include/CGAL/poisson_refine_triangulation.h [hunk 3]
```diff
@@ -180,17 +180,17 @@ class Poisson_mesher_level :
...
+/// *Preconditions:*
 /// - Tr must use a geometric traits with robust circumcenter computation.
 /// - convergence is guaranteed if radius_edge_ratio_bound >= 1.0.
 ///
-/// @commentheading Template Parameters:
-/// @param Tr 3D Delaunay triangulation.
-/// @param Surface Sphere_3 or Iso_cuboid_3.
-/// @param Sizing_field A sizing field functor type
-/// @param Second_sizing_field A sizing field functor type
+/// *Template Parameters*
```
Comment (albert-github) [inline Poisson_surface_reconstruction_3/include/CGAL/poisson_refine_triangulation.h:187]: Is this line not a bit overdone as doxygen will add a header **Template Parameters**

File: Poisson_surface_reconstruction_3/include/CGAL/poisson_refine_triangulation.h [hunk 4]
```diff
@@ -180,17 +180,17 @@ class Poisson_mesher_level :
 /// bad means badly shaped or too big).
 /// @return the number of vertices inserted.
 ///
-/// @commentheading Preconditions:
+/// *Preconditions:*
```
Comment (albert-github) [inline Poisson_surface_reconstruction_3/include/CGAL/poisson_refine_triangulation.h:183]: Why not use the doxygen `\pre` command?

File: Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h [hunk 5]
```diff
@@ -348,12 +348,12 @@ class Reconstruction_triangulation_3
   /// Insert the [first, beyond) range of points in the triangulation using a spatial sort.
   /// Default type is INPUT.
   ///
-  /// @commentheading Template Parameters:
-  /// @param InputIterator iterator over input points.
-  /// @param PointPMap is a model of `ReadablePropertyMap` with a value_type = Point_3.
+  /// *Template Parameters:*
```
Comment (albert-github) [inline Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h:351]: Is this line not a bit overdone as doxygen will add a header **Template Parameters**

File: Polyhedron/include/CGAL/Polyhedron_3.h (hunk 1 above)
Comment (afabri) [reply Polyhedron/include/CGAL/Polyhedron_3.h:1451]: thanks. I'll change.

**Pull request merged**